test/test_broker.hpp

```diff
@@ -997,11 +1017,26 @@ class test_broker {
             BOOST_ASSERT(non_active_sessions_.get<tag_client_id>().count(client_id) == 0);
             BOOST_ASSERT(non_active_sessions_.get<tag_client_id>().find(client_id) == non_active_sessions_.get<tag_client_id>().end());
 
+            auto const& sei_opt = state->session_expiry_interval;
+            if (sei_opt && sei_opt.value() != std::chrono::seconds(0xffffffffUL)) {
+                state->tim_session_expiry.expires_after(sei_opt.value());
+                state->tim_session_expiry.async_wait(
+                    [&, wp = session_state_wp(state)]
```

Comment:
This doesn't need to be shared_ptr / weak_ptr.

The lambda can store the connection shared pointer, that's sufficient to look up the active session state object.

Reply:
You clear session_state::con at https://github.com/redboltz/mqtt_cpp/pull/657/files#diff-45ca9626e6c7c611be2ceb2ff63031a5L988 and move it to non_active_sessions.

Reply:
I'm not sure removing clearing `con` code is good. Also I don't know why you add clearing `con` code.
I guess that some of resources are released by the endpoint destructor and it should be happened before insert non_active_sessions.
So I didn't modify the clearing code.